src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,174 @@
+/*
```

Comment:
It's confusing to have both "server" and "ui" in the file name. Would it be better to rename it to "storyTellingUiHandler" or "storyTellingUiFetcher"?

It also looks redundant to have almost all files in src/ui/ directory starting with "storyTelling" which make file names longer. The project name has been clear about "story-telling".

Reply:
I agree on all counts. The files in the `ui` directory were named roughly according to their grade names in order to help make finding the relevant files easier, though this naming convention is out of keeping with the files in the `server` directory. This is, more than anything else, simply a remnant of the way we kicked off the project with a few days of quick-and-dirty development and a fair amount of premature optimization. I will remove "storyTelling." from all file names and think of a more appropriate name for `storyTellingServerUI.js`.

Reply:
In a related issue, I think it would make sense at some point to review all grade names and make sure they still make sense, given that the project has changed so much in the last two years

Reply:
Re `storyTellingServerUI.js`, I think it might need more than just a renaming. It might make more sense to refactor these functions a bit and put them in other files. `loadBrowse` and `loadStoryFromParameter`, in addition to possibly being renamed themselves, would more logically live in the same file as the `storyBrowse` and `storyView` grades they end up calling. Also my usage of promises and callback parameters simultaneously isn't ideal, so this might be a good opportunity to switch to an all-promises setup.

Reply:
What a coincident that I've been thinking about refactoring of this file too. I actually think all functions in this file can be moved into components and all workflows can be incorporated and controlled by individual components. Use "Browse story" as an example:

1. move the fetch of "/clientConfig" into "sjrk.storyTelling.base.page" `onCreate` event;
2. The retrieval of stories to "/stories" endpoint can be moved to  "sjrk.storyTelling.base.page.storyBrowse" component, in parallel with the fetch of "/clientConfig". It also fires an event "onStoriesReady" when stories are received.
3. The aggregate event for`onClientConfigReady` + "onStoriesReady" controls other parts such as the load of custom theme files, the instantiation of templateManager and the page rendering.

Considering our timeline, it might be better to do this improvement in a separate JIRA if we decide to do it.

Reply:
[SJRK-265](https://issues.fluidproject.org/browse/SJRK-265) filed to address this